packages/angular/ssr/src/app.ts

```diff
@@ -134,45 +131,45 @@ export class AngularServerApp {
       }
     }
 
-    return this.handleAbortableRendering(
-      request,
-      /** isSsrMode */ true,
-      matchedRoute,
-      requestContext,
-    );
-  }
-
-  /**
-   * Retrieves the matched route for the incoming request based on the request URL.
-   *
-   * @param request - The incoming HTTP request to match against routes.
-   * @returns A promise that resolves to the matched route metadata or `undefined` if no route matches.
-   */
-  private async getMatchedRoute(request: Request): Promise<RouteTreeNodeMetadata | undefined> {
-    this.router ??= await ServerRouter.from(this.manifest, new URL(request.url));
-
-    return this.router.match(new URL(request.url));
+    return Promise.race([
+      new Promise<never>((_, reject) => {
+        request.signal.addEventListener(
+          'abort',
+          () => {
+            const abortError = new Error(
+              `Request for: ${request.url} was aborted.\n${request.signal.reason}`,
+            );
+            abortError.name = 'AbortError';
+            reject(abortError);
+          },
+          { once: true },
+        );
```

Comment:
Consider: Move this to a helper function which creates a `Promise` from an `AbortSignal`.